test/addons/esm/test.js

```diff
@@ -0,0 +1,9 @@
+// Flags: --experimental-addon-modules
+
+// FIXME(legendecas): AddonTestConfiguration does not support '.mjs' extension.
+// Eslint complains about ESM syntax in a file with '.js' in 'test/addons'.
+'use strict';
+const common = require('../../common');
+
+import('./test-esm.mjs')
+  .then(common.mustCall(), common.mustNotCall());
```

Comment:
FWIW the `common.mustNotCall()` is not necessary. I assume we also want to test if loading it with `require` works
```suggestion
import('./test-esm.mjs').then(common.mustCall());

require('./test-esm.mjs');
```